### DISCUSSION TITLE
License of ImageMagick and 3rd party licenses

### DISCUSSION THREAD

**Discussion by MMcKester** [ImageMagick/ImageMagick#2498] [Help] [answered]
Hi, I am reaching out because I have a question regarding the license of ImageMagick for my commercial application I would like to release in the future. ImageMagick itself is licensed under the "ImageMagick License". But that doesn't count for the dependencies, means I have to comply with each of them as well, is that correct?

As far as I can see ImagicMagick depends on FFMpeg, GhostScript, Libsrvg, etc. Does that mean I have to verify each dependencies license as well? In my case I don't link ImageMagick into my main application, it's more used in an external helper process which performs a certain task. And this helper process would be open source. Is that a correct approach?

I know, this topic requires an attorney, but any help or insights are appreciated!

**Comment by urban-warrior** [comment]
ImageMagick has an Apache II license which is quite liberal per https://imagemagick.org/script/license.php.  We purposefully utilize delegates as not to pollute the license.  Some portions of ImageMagick include source from other projects that do not diminish the ImageMagick license, i.e. either public domain or BSD which are more liberal than the Apache II license.  For the delegate libraries and programs, you are on your own.  We can only assist with ImageMagick.

**Reply by MMcKester** [reply]
Thank you very much for your answer!

I would like to bundle the "[ImageMagick-7.0.10-28-portable-Q16-HDRI-x64.zip](https://imagemagick.org/script/download.php)" from the download section with my commercial application - no linking involved - just calling e.g. the "convert.exe" by command line through my app.
 
The package itself looks like it supports a lot, but the license.txt of that package does not indicate that there are any other external libraries used, is that correct? So I am wondering if it would be ok if I would ship the content of the zip file with my app?

**Comment by urban-warrior** [comment, marked as answer]
Of course you can ship the content of the zip file with your app-- as long as you comply with the licenses.  Read NOTICE.txt inside the distribution for the third-party licenses.  Run `magick --version` to identify which third-party delegate libraries are included in the distribution.  Some of the delegate programs included in the distribution are ffmpeg, dcraw, and hp2xx.exe.  These can be removed if you don't agree with the license and ImageMagick will still function with diminished capacity.

**Reply by MMcKester** [reply]
Thank you so much! That is exactly what I was looking for! :-)
